src/tscore/MatcherUtils.cc

```diff
@@ -618,3 +618,75 @@ parseConfigLine(char *line, matcher_line *p_line, const matcher_tags *tags)
 
   return nullptr;
 }
+
+// char* parseYamlDoc(YAML::Node node, matcher_line* p_line, const matcher_tags* tags)
+//
+//   Parse out a yaml node object suitable for passing to
+//    a ControlMatcher object
+//
+//   If successful, nullptr is returned.  If unsuccessful,
+//     a static error string is returned
+//
+const char *
+parseYamlDoc(matcher_line *p, const matcher_tags *tags)
+{
+  const char *label;
+  const char *val;
+  matcher_type type = MATCH_NONE;
+
+  YAML::Node root = *(p->root_node);
```

Comment:
Allright. The `matcher_line` contains a `root_node` which is presumed to be a list of objects, each of which has keys that correspond to the keys in `tags`. All the instances of `p` passed in point to the same root node? Why can't the `matcher_line` just point directly at the object?